### ISSUE TITLE
Template compiling error with module-unification blueprint

### ISSUE THREAD

**Issue by phammers** [ember-cli/ember-cli#8049]
I'm trying to upgrade my application from 3.3 to 3.4.  The application is using pods - I'm unsure if that's related. It seems to be failing trying to compile the module-unification blueprint.  If I remove that directory from my node_modules directory, the build works fine.

```
ERROR Summary:

  - broccoliBuilderErrorStack: Error: Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0
    at new SyntaxError (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2858:31)
    at TokenizerEventHandlers.HandlebarsNodeVisitors.MustacheStatement (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:3130:27)
    at TokenizerEventHandlers.Parser.acceptNode (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2950:35)
    at TokenizerEventHandlers.HandlebarsNodeVisitors.Program (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:3060:22)
    at TokenizerEventHandlers.Parser.acceptNode (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2950:35)
    at preprocess (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:4175:56)
    at exports.precompile (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2567:42)
    at Object.exports.default (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:6265:37)
    at Object.template (/home/vagrant/projects/csa.ui/node_modules/ember-cli-htmlbars/utils.js:29:40)
    at TemplateCompiler.processString (/home/vagrant/projects/csa.ui/node_modules/ember-cli-htmlbars/index.js:77:40)
  - codeFrame: Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0
  - errorMessage: Build Canceled: Broccoli Builder ran into an error with `broccoli-persistent-filter:TemplateCompiler` plugin. 💥
Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0
  - errorType: Template Compiler Error
  - location:
    - column: 0
    - file: vendor/ember-cli/blueprints/module-unification-app/files/src/ui/routes/application/template.hbs
    - line: 6
    - treeDir: /home/vagrant/projects/csa.ui/tmp/broccoli_persistent_filtertemplate_compiler-input_base_path-JepdXZHY.tmp
  - message: Build Canceled: Broccoli Builder ran into an error with `broccoli-persistent-filter:TemplateCompiler` plugin. 💥
Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0
  - name: Error
  - nodeAnnotation: broccoli-persistent-filter:TemplateCompiler
  - nodeName: broccoli-persistent-filter:TemplateCompiler
  - originalErrorMessage: Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0
  - stack: Error: Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0
    at new SyntaxError (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2858:31)
    at TokenizerEventHandlers.HandlebarsNodeVisitors.MustacheStatement (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:3130:27)
    at TokenizerEventHandlers.Parser.acceptNode (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2950:35)
    at TokenizerEventHandlers.HandlebarsNodeVisitors.Program (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:3060:22)
    at TokenizerEventHandlers.Parser.acceptNode (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2950:35)
    at preprocess (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:4175:56)
    at exports.precompile (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:2567:42)
    at Object.exports.default (/home/vagrant/projects/csa.ui/node_modules/ember-source/dist/ember-template-compiler.js:6265:37)
    at Object.template (/home/vagrant/projects/csa.ui/node_modules/ember-cli-htmlbars/utils.js:29:40)
    at TemplateCompiler.processString (/home/vagrant/projects/csa.ui/node_modules/ember-cli-htmlbars/index.js:77:40)
```

---

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 3.4.2
http_parser: 2.8.0
node: 8.11.2
v8: 6.2.414.54
uv: 1.19.1
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 57
nghttp2: 1.29.0
napi: 3
openssl: 1.0.2o
icu: 60.1
unicode: 10.0
cldr: 32.0
tz: 2017c
os: linux x64
5.6.0
1.7.0

```

**Comment by rwjblue** [comment]
> Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0

The error message seems pretty explicit about what is wrong here, doesn’t it? Presumably you have `<{{outlet}}` at the beginning of line 6...

**Comment by phammers** [comment]
@rwjblue thank you for your response.  Yes, the error is very descriptive, but it doesn't seem to be in our application template - it's popping in the ember-cli blueprint here: `node_modules/ember-cli/blueprints/module-unification-app/files/src/ui/routes/application/template.hbs`.

When I remove that file everything compiles fine.

**Comment by dfreeman** [comment]
One of our devs ran into this in a project and we tracked it down in our case to `ember-cli-iframe-resizer`, which [effectively puts all of `node_modules` in its `vendor` tree](https://github.com/selvaa89/ember-cli-iframe-resizer/blob/master/index.js#L10).

For anyone else who hits this issue, removing dependencies from your `package.json` until your build suddenly works may be a viable way to figure out which addon is misbehaving.

**Comment by jenweber** [comment]
I encountered this issue when I had a third-party component UI library duplicated in 2 different places. It was in the package.json within my own addon (v2.16) and in an app using my own addon (v3.7). When we upgraded the internal addon to v3.7 is when we saw this error.

The third-party addon was https://github.com/Semantic-Org/Semantic-UI-Ember but my guess is that would happen for other component libraries as well, and it's not just a problem with this addon.

**Comment by jenweber** [comment]
TLDR: redundant addons and file imports cause weird problems.

I also experienced an issue that only popped up when using `npm link`. For example, I have an addon that uses some css in a folder called `semantic`, and that same css is duplicated in the vendor folder. Initially I had the following in the consuming app's `ember-cli-build.js`:

```
// bad, broken
app.import('node_modules/ember-biobright-core/semantic/dist/semantic.css')
```

Fixed:
```
// good
app.import('vendor/semantic/dist/semantic.css')
```

For whatever reason, the "bad" version worked ok when installed via npm/GitHub and _not_ when served using `npm link` to associate the package.

These comments above are purely for people trying to debug this error in their own apps. These errors are very edge-casey that I don't think it's a problem with Ember CLI.

**Comment by irminsul** [comment]
I encountered this issue as well during the upgrade of an old Ember project. In my case it was an old `.bowerrc` file in the project's root directory.

Minimal reproduction:

```shell-session
$ ember new --yarn issue-8049
$ cd issue-8049
$ echo '{ "directory": "node_modules" }' > .bowerrc
$ ember build
Environment: development
cleaning up...
Template Compiler Error (broccoli-persistent-filter:TemplateCompiler) in node_modules/ember-cli/blueprints/module-unification-app/files/src/ui/routes/application/template.hbs

Cannot use mustaches in an elements tagname: `{{outlet` at L6:C0


Stack Trace and Error Report: /tmp/error.dump.2dda1a8302ad60cd2d9a397e89b5edf7.log
```

**Comment by Turbo87** [comment]
closing due to https://blog.emberjs.com/2019/03/11/update-on-module-unification-and-octane.html

**Comment by ballPointPenguin** [comment]
Note: I have been seeing the same error on an older ember build. In my case the addon ember-cli-selectize was the culprit.

**Comment by ballPointPenguin** [comment]
I encountered this issue in an older app and after much trial-and-error discovered that in my case it was resolved by removing `ember-cli-ic-ajax` from my package.json
